app_server/app/controllers/SubscriptionController.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Subscription Controller
+ * @module SubscriptionController
+ */
+var rfr = require('rfr');
+var Joi = require('joi');
+var Boom = require('boom');
+var Promise = require('bluebird');
+
+var CustomError = rfr('app/util/Error');
+var Utility = rfr('app/util/Utility');
+var Service = rfr('app/services/Service');
+var Authenticator = rfr('app/policies/Authenticator');
+
+var logger = Utility.createLogger(__filename);
+
+function SubscriptionController(server, options) {
+  this.server = server;
+  this.options = options;
+}
+
+var Class = SubscriptionController.prototype;
+
+Class.registerRoutes = function() {
+
+  this.server.route({method: 'POST', path: '/{id}',
+                     config: {
+                       validate: singleSubscriptionValidator,
+                       auth: {scope: Authenticator.SCOPE.ALL}
+                     },
+                     handler: this.createSubscription});
+
+  this.server.route({method: 'GET', path: '/', // private
+                     config: {
+                       auth: {scope: Authenticator.SCOPE.ALL}
+                     },
+                     handler: this.getSubscriptions});
+
+  this.server.route({method: 'GET', path: '/{id}',
+                     config: {
+                       auth: {scope: Authenticator.SCOPE.ALL}
+                     },
+                     handler: this.getSubscribers});
+
+  this.server.route({method: 'DELETE', path: '/unsubscribe/{id}',
+                     config: {
+                       validate: singleSubscriptionValidator,
```

Comment:
The route's paths here don't conform with with REST API's design in which a route should be a path to a `resource` and contains no action with partial exceptions for APIs for controlling streams and searching. The actions are reflected by the HTTP method.

So the route's paths that have to do with subscribers here can be prepended with `/subscribers/...` instead of an ambiguous route (in `line39`) or actions like `unsubscribe`.